vms/platformvm/txs/mempool/mempooltest/mempool.go

```diff
@@ -0,0 +1,133 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package mempooltest
+
+import (
+	"context"
+	"sync"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow/engine/common"
+	"github.com/ava-labs/avalanchego/utils/lock"
+	"github.com/ava-labs/avalanchego/utils/set"
+	"github.com/ava-labs/avalanchego/vms/platformvm/txs"
+	"github.com/ava-labs/avalanchego/vms/platformvm/txs/mempool"
+)
+
+var _ mempool.Mempool = (*FakeMempool)(nil)
+
+// FakeMempool performs no verification and has no upper bound on txs
+type FakeMempool struct {
+	txs  map[ids.ID]*txs.Tx
+	drop map[ids.ID]error
+
+	mu   sync.Mutex
+	cond lock.Cond
+}
+
+func (f *FakeMempool) initTxs() {
+	if f.txs != nil {
+		return
+	}
+
+	f.txs = make(map[ids.ID]*txs.Tx)
+}
+
+func (f *FakeMempool) Add(tx *txs.Tx) error {
+	f.mu.Lock()
+	defer f.mu.Unlock()
+
+	f.txs[tx.ID()] = tx
+	return nil
+}
+
+func (f *FakeMempool) Get(txID ids.ID) (*txs.Tx, bool) {
+	f.mu.Lock()
+	defer f.mu.Unlock()
+
+	f.initTxs()
+
+	tx, ok := f.txs[txID]
+	return tx, ok
+}
+
+func (f *FakeMempool) GetDropReason(txID ids.ID) error {
+	f.mu.Lock()
+	defer f.mu.Unlock()
+
+	if f.drop == nil {
+		f.drop = make(map[ids.ID]error)
+	}
+
+	err := f.drop[txID]
+	return err
```

Comment:
We shouldn't need to init the map here.
```suggestion
	return f.drop[txID]
```